### PR TITLE
refactor(lane_change): move several functions from lane change utils to NormalLaneChange

### PR DESCRIPTION
**Pull request by purewater0901**
## Description
In order to reduce the arguments in the `getPrepareSegment`, `hasEnoughLength` and `getTargetObjects`, I moved them to the `NormalLaneChange` class. Therefore, it has two less arguments than before. Moreover, I removed a single `maybe unused` flag to clean the code.

## Tests performed

1577/1588
[TIER IV Internal Link](https://evaluation.tier4.jp/evaluation/reports/aeee5948-c143-52b5-be1a-584ac0d2c446?project_id=prd_jt)

## Effects on system behavior

No effects.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp [hunk 1]
```diff
@@ -482,6 +482,141 @@ PathWithLaneId NormalLaneChange::getPrepareSegment(
   return prepare_segment;
 }
 
+LaneChangeTargetObjects NormalLaneChange::getTargetObjects(
+  const PredictedObjects & objects, const lanelet::ConstLanelets & current_lanes,
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp:486]: @purewater0901 
I would recommend just remove parameter `objects` and just call `planner_data->dynamic object directly in this function.

File: planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp [hunk 2]
```diff
@@ -482,6 +482,141 @@ PathWithLaneId NormalLaneChange::getPrepareSegment(
...
+  const lanelet::ConstLanelets & target_lanes,
+  const lanelet::ConstLanelets & target_backward_lanes) const
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp:488]: If you want to reduce further,I would also recommend `target_backward_lanes` to be computed in this function.

File: planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp (hunk 1 above)
Comment (purewater0901) [reply planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp]: done with https://github.com/autowarefoundation/autoware.universe/pull/4273/commits/82c4c8c2787bd902aef7d3e28693971e1f6a4935

File: planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp (hunk 2 above)
Comment (purewater0901) [reply planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp]: done with https://github.com/autowarefoundation/autoware.universe/pull/4273/commits/82c4c8c2787bd902aef7d3e28693971e1f6a4935

**Review by zulfaqar-azmi-t4 — APPROVED** [review]
LGTM!

**Pull request merged**